src/MongoDB.Driver.Core/Core/Servers/Server.cs

```diff
@@ -54,13 +54,14 @@ internal sealed class Server : IClusterableServer
         #endregion
 
         // fields
+        private readonly ServerDescription _baseDescription;
```

Comment:
For me, this logic is not really for Server.cs, I guess we can make SetDescription public in serverMonitor and call this method instead of this line?